lib/server/swarm.js

```diff
@@ -7,12 +7,27 @@ var randomIterate = require('random-iterate')
 // Regard this as the default implementation of an interface that you
 // need to support when overriding Server.createSwarm() and Server.getSwarm()
 function Swarm (infoHash, server) {
+  this.complete = 0
+  this.incomplete = 0
   this.peers = new LRU({
     max: server.peersCacheLength || 1000,
-    maxAge: server.peersCacheTtl || 900000 // 900 000ms = 15 minutes
+    maxAge: server.peersCacheTtl || 20 * 60 * 1000 // 20 minutes
+  })
+
+  // When a websocket peer is evicted from the LRU cache, close the websocket
+  // after a short timeout period. We wait 1s so the server has a chance to send
+  // a response to 'stopped' events, which remove the peer and cause an eviction.
+  this.peers.on('evict', function (data) {
+    var peer = data.value
+    if (peer.socket) {
+      setTimeout(function () {
+        try {
+          peer.socket.close()
```

Comment:
> LRU .remove method does not emit evict.

Good point. I needed the setTimeout to get the tests to pass, but it looks like I misunderstood what `remove()` was doing. I'll look again.

> We should perform the stuff done in the event of stopped message when the eviction happens.

That's what happens when we call `socket.close()`. The `onWebSocketClose()` function will run, and that does a simulated 'stopped' event.